express-notes-api/delete.js

```diff
@@ -0,0 +1,23 @@
+module.exports = function deleteEntry(app, data, path, fs, writeForDelete) {
+  app.delete('/api/notes/:id', (req, res) => {
+    const id = req.params.id;
+    const selectedNote = data.notes[id];
+    delete data.notes[id];
```

Comment:
It's best to do your error checking before you make any changes to the data. That way the data will remain in good shape if you end up rejecting the request. It is good, though, that you don't write out the data if there's a request error!